### DISCUSSION TITLE
Commit issue

### DISCUSSION THREAD

**Discussion by jishkln** [community/community#30005] [Other Feature Feedback, Questions, & Ideas]
I'm using Ubuntu , in each and every commit asking the user name and the token, how to fix it.

**Comment by aayushchugh** [comment]
You should setup `SSH` key for your GitHub account [docs](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)

**Comment by airtower-luna** [comment]
There's nothing *broken* to fix, Git just asks for credentials when it needs them. But I get it's annoying. :sweat_smile: 

So to avoid it you can tell Git to keep them, or (as mentioned above) use SSH where the SSH Agent can keep your key for a while. If you want to stick with HTTPS, check [Git Tools - Credential Storage](https://git-scm.com/book/en/v2/Git-Tools-Credential-Storage). The example there shows to set up the `cache` helper, which keeps your credentials in RAM for a while. Long-term options exist, too, if they fit your security needs.

In my opinion keeping a token in a password manager and using `cache` so you don't have to provide it all the time is a good compromise.
